File: src/C/inserts.cc
#include "_common.h"

void do_benchmark_inserts(long int count) {
  long int i = 0;
  const char* query = cfg["insert_query"].c_str();

  for (i = 0; i < count; i++)
    mysql_query(conn, query);
}

int main(int argc, char *argv[]) {
  long int count;

  parseCfg(argc, argv);

  do_init(0);

  count = atol(cfg["insert_rows_count"].c_str());
  do_benchmark(do_benchmark_inserts, "inserts", count);

  mysql_close(conn);

  return 0;
}

File: src/PHP/inserts.php
#!/usr/bin/env php
<?php

include_once("_common.php");

function do_benchmark_inserts() {
  global $conn, $cfg;

  $start = microtime(true);

  for ($i = 0; $i < $cfg['insert_rows_count']; $i++)
    mysql_query($cfg['insert_query'], $conn);

  $finish = microtime(true);

  return round($cfg['insert_rows_count'] / ($finish - $start), 0);
}

do_init();

$results = array();

$results = do_benchmark_inserts();

mysql_close($conn);

echo json_encode($results);
